009-OFEP-add-dispose.md

```diff
@@ -0,0 +1,42 @@
+## Add dispose functionality to API
+
+## State: DRAFTING
+
+The goal of this OFEP is to enhance the OpenFeature API to allow authors to cleanly dispose of any resources by consumed by OpenFeature providers.
+
+## Background
+
+When implementing OpenFeature providers, you sometimes need to register event handlers, or timers or similar functionality to
+correctly handle functionality of the vendor library. Some vendor libraries might also have analytics, or debugging functionality
+that can be enabled and might get flushed to the vendor platform when the library gets stopped or disposed off.
+
+Currently, you aren't able to cleanly handle these use cases in OpenFeature providers.
+
+## Proposal
+
+To allow to cleanly dispose of resources consumed by OpenFeature providers, I would like to purpose a `dispose`-function that
+can be called by OpenFeature SDK to allow to cleanup these resources. The OpenFeature provider will implement an async function
+named `dispose` that can be called by the internal OpenFeature SDK.
```

Comment:
Should we have a comment here for idiomatic alternatives? "`dispose` or technology idiom equivalent"? However similar things have been worded in the spec.

For instance, in .Net we should likely do this by implementing `IDisposable` (which happens to be the same name for the method).  

Also, being async would probably not be appropriate for some languages. So maybe `async` when reasonable?

Reply:
Yeah, I think it's reasonable to write down something like what the best practice is for the chosen programming language? E.g. `IDisposable` in .Net, `Destroy` called by `Free` in Delphi etc.

Reply:
> Should we have a comment here for idiomatic alternatives? "dispose or technology idiom equivalent"? However similar things have been worded in the spec.

I'll add wording like you describe here as well.

> Also, being async would probably not be appropriate for some languages. So maybe async when reasonable?

Agreed.

Reply:
I've made this update.